modules/admin-ui-picard/app/src/components/events/partials/modals/EmbeddingCodeModal.js

```diff
@@ -0,0 +1,144 @@
+import React, {useEffect, useState} from 'react';
+import {useTranslation} from "react-i18next";
+import axios from "axios";
+
+
+/**
+ * This component renders the embedding code modal
+ */
+const EmbeddingCodeModal = ({ close, eventId }) => {
+    const { t } = useTranslation();
+
+    const [textAreaContent, setTextAreaContent] = useState('');
+    const [sourceURL, setSourceURL] = useState('');
+    const [currentSize, setCurrentSize] = useState('0x0');
+    const [showCopySuccess, setCopySuccess] = useState(false);
+
+    useEffect(() => {
+        // get source url
+        axios.get("/api/info/organization/properties/engageuiurl")
+            .then(res => {
+            let url = res.data["org.opencastproject.engage.ui.url"];
+            setSourceURL(url);
+        }).catch(res => {
+            setSourceURL("<SERVER_URL>");
+        });
```

Comment:
I would leave this here since a thunk normally dispatches an action/change in the redux state. I could move it to an extra file like a utils file or something if this is ok?

Reply:
That's okay, I just thought the API requests shouldn't be in the same file, as the modal. The alternative would have been to move the state of the modal into the redux state and move the api request into a thunk.